src/components/tables/Table.tsx

```diff
@@ -0,0 +1,360 @@
+            tableId={this.props.id}
+            isCollapsible={!!collapsibleData}
+            onClickCallback={(e: React.MouseEvent<any>) =>
+              this.props.onRowClick(this.props.getActions && this.props.getActions(rowData, this.props))
+            }>
+            {tableHeadingRowContent}
+          </TableHeadingRowConnected>
+          {collapsibleRow}
+        </TableRowWrapper>
+      );
+    });
+  }
+
+  private buildBlankSlate(): JSX.Element {
+    const { tableCompositeState } = this.props;
+    const tableData = tableCompositeState.data || this.props.initialTableData;
+    const {
+      blankSlateDefault,
+      blankSlateNoResultsOnAction,
+      blankSlateOnError,
+    } = this.props;
+
+    let blankSlatePropsToUse: IBlankSlateProps;
+
+    if (tableData.displayedIds.length
+      || this.props.tableCompositeState.isLoading) {
+      return null;
+    }
+
+    if (tableCompositeState.filter || _.some(tableCompositeState.predicates, (value: any) => !_.isUndefined(value))) {
+      blankSlatePropsToUse = blankSlateNoResultsOnAction || blankSlateDefault;
+    } else if (tableCompositeState.isInError) {
+      blankSlatePropsToUse = blankSlateOnError || blankSlateDefault;
+    } else {
+      blankSlatePropsToUse = blankSlateDefault;
+    }
+
+    return <BlankSlate {...blankSlatePropsToUse} />;
+  }
+
+  private buildNavigation(): JSX.Element {
+    const tableData = this.props.tableCompositeState.data || this.props.initialTableData;
+
+    return !!this.props.navigation ? (
+      <NavigationConnected
+        {...this.props.navigation}
+        totalEntries={tableData.totalEntries}
+        totalPages={tableData.totalPages}
+        id={getTableChildComponentId(this.props.id, TableChildComponent.NAVIGATION)}
+        loadingIds={[getTableChildComponentId(this.props.id, TableChildComponent.LOADING_NAVIGATION)]}
+      />
+    ) : null;
+  }
+
+  private buildLastUpdated(): JSX.Element {
+    return <LastUpdatedConnected
+      label={this.props.lastUpdatedLabel}
+      id={getTableChildComponentId(this.props.id, TableChildComponent.LAST_UPDATED)} />;
+  }
+
+  render() {
+    const tableClasses = classNames(
+      'mod-collapsible-rows',
+      'mod-align-header',
+      {
+        'mod-loading-content': !!(this.props.tableCompositeState && this.props.tableCompositeState.isLoading),
+        'loading-component': this.isInitialLoad,
+      },
+    );
+
+    return (
+      <div className='table-container'>
+        {this.buildActionBar()}
+        <table className={tableClasses}>
+          <tbody className='loading-row'>
+            <tr>
+              <td colSpan={this.props.headingAttributes.length + TOGGLE_ARROW_CELL_COUNT}>
+                <Loading />
+              </td>
+            </tr>
+          </tbody>
+          {this.buildTableHeader()}
+          {this.buildTableBody()}
+        </table>
+        {this.buildBlankSlate()}
+        {this.buildNavigation()}
+        {this.buildLastUpdated()}
```

Comment:
This file is getting very big. How about we move all of the `buildSomething` as const with props in other files and passing the needed props? This way the file will be smaller and we will already have the logic if we want to do something similar elsewhere.